### ISSUE TITLE
Update GPX library

### ISSUE THREAD

**Issue by tonymorris** [tonymorris/geo-gpx#1]
I am considering updating this library. In particular;
- Use `Control.Lens` instead of `Data.Lens`
- Implement zipper
- Implement tests against GPX files
- Implement tests using `doctest`
- Read whitespace in GPX file so as to recover the original file.
- Rename the package from `GPX` to `gpx` (what is the procedure for this?)
- Provide higher-level functions for easy manipulation of gpx files.
- Use `Data.Text` instead of `String`
- Use explicit imports and `NoImplicitPrelude` (code style)
- User a proper coordinate data type

Just wondering if there might be any other interest or comment.

@TomMD

**Comment by ocharles** [comment]
I would be interested in helping out with some of this, as I have a need to use this package. Specifically, I would be willing to:
- Use `Control.Lens`
- Use `Data.Text`

I'm wondering about exposing the constructors more, too. I also think the package would be a lot easier to navigate if there were less modules.

**Comment by tonymorris** [comment]
Yeah I agree with exposing the constructors where appropriate (i.e. hiding is not to enforce some sort of invariant).

**Comment by TomMD** [comment]
@tonymorris I appreciate the ping on this, but I don't have the bandwidth to contribute at this time.

Edit: Just noticed the ping was years ago... oh well.
